controllers/multiclusterobservability/multiclusterobservability_controller_test.go

```diff
@@ -298,6 +298,11 @@ func TestMultiClusterMonitoringCRUpdate(t *testing.T) {
 	// Create a ReconcileMemcached object with the scheme and fake client.
 	r := &MultiClusterObservabilityReconciler{Client: cl, Scheme: s, OcpClient: ocpClient, CrdClient: crdClient}
 	config.SetMonitoringCRName(name)
+
+	err = cl.Create(context.TODO(), mco)
```

Comment:
What is different between init object with call `create` explicitly?

Reply:
This change doesn't matter actually based on my experiments.
The one matters is the MCO CR delete option, currently the MCO CR deletion will be returned soon, but the MCO CR is actually still there because the finalizer is not cleaned.

I tried to add a reconcile loop after the deletion operator, but the reconcile loop hang somewhere. I guess it may be related to the status update goroutine, just my guess.

Currently, the most simple fix is to delete the finalizer manually after the MCO CR deletion operaton, then the MCO CR can be successfully deleted.